src/olympia/addons/templates/addons/impala/details.html

```diff
@@ -322,28 +322,30 @@ <h4 class="author">{{ _('by') }} {{ users_list(addon.listed_authors) }}</h4>
       </div>
     </div>
   {% endif %}
-  {% if waffle.switch('webext-permissions') and addon.current_version and addon.current_version.all_files[0] %}
+  {% if waffle.switch('webext-permissions') and
+        addon.current_version and addon.current_version.all_files[0] and
+        (not addon.current_version.all_files[0].is_webextension or
+         addon.current_version.all_files[0].webext_permissions)
```

Comment:
yeah... I was going to do that but couldn't decide where was best to stick the method (should be on File but by that point you already have to add 3 conditions).  Addon is as good a place as any I guess.